src/app/@modal/(...)courses/[slug]/manage/page.tsx

```diff
@@ -0,0 +1,62 @@
+import { notFound, redirect } from 'next/navigation'
+
+// import { slugifyCourseName, urlForCourse } from '@lib/courses'
+import { getCourseInfo } from '@services/archive'
+
+import Modal from '@components/Modal'
+import RenameCourse from '@components/tools/RenameCourse'
+import DeleteCourse from '@components/tools/DeleteCourse'
+import { validateRights } from '@services/tkoUserService'
+
+const parseSlug = (slug: string) => {
+  const parsedSlug = slug.match(/(?<id>\d+)-(?<courseSlug>.*)/)
+  if (!parsedSlug || !parsedSlug.groups) {
+    notFound()
+  }
+
+  const id = parseInt(parsedSlug.groups.id, 10)
+
+  if (isNaN(id)) {
+    notFound()
+  }
+
+  return {
+    id,
+    courseSlug: parsedSlug.groups.courseSlug
+  }
+}
+
+const Page = async ({ params }: any) => {
+  const isRights = await validateRights('rename', 'remove')
+
+  if (!isRights) {
+    redirect('/')
+  }
+
+  const { id, courseSlug } = parseSlug(params.slug)
+
+  const course = await getCourseInfo(id)
+  if (!course) {
+    notFound()
+  }
+
+  // // if (courseSlug !== slugifyCourseName(course.name)) {
```

Comment:
Why do we not do this?

Reply:
If redirecting is back, we could at least set the window location to visually update it (and allow for consistent URL sharing)

Reply:
This has been commented out as I also want this to be working, but this were causing previously redirect loop without reason. TODO: test if this still causes some issues if enabled.

Reply:
Brought back as Next's app router has been developed further and not causing problems anymore.